src/libvirt-xml-update.js

```diff
@@ -296,3 +296,56 @@ export function updateMaxMemory(domXml, maxMemory) {
 
     return s.serializeToString(doc);
 }
+
+function xmlGetXPath(doc, xpath) {
+    const elements = [];
+    let node = null;
+    const iterator = doc.evaluate(xpath, doc);
+    while ((node = iterator.iterateNext()))
+        elements.push(node);
+    return elements;
+}
+
+export function replaceSpice(domXml) {
+    // this implements https://access.redhat.com/solutions/6955095
+    logDebug("replaceSpice original XML:", domXml);
+
+    const doc = getDoc(domXml);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1387-20240122-073835-c9864bc3-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)